spatialprofilingtoolbox/ondemand/providers/counts_provider.py

```diff
@@ -0,0 +1,103 @@
+"""Count cells for a specific signature, over the specially-created binary-format index."""
+
+from typing import Any
+
+from spatialprofilingtoolbox.ondemand.providers import Provider
+from spatialprofilingtoolbox.standalone_utilities.log_formats import colorized_logger
+
+logger = colorized_logger(__name__)
+
+
+class CountsProvider(Provider):
+    """Scan binary-format expression matrices for specific signatures."""
+
+    def __init__(self, data_directory: str, load_centroids: bool = False) -> None:
+        """Load from a precomputed JSON artifact in the data directory.
```

Comment:
Is leading triple quotes like this the recommended style?
I thought in blocks you are supposed to bound the docstring above and below the same way, unless it is a single line.

Reply:
I believe putting the one liner as close to the definition as possible is the recommended style.

Reply:
Hm most of the examples in PEP8 and PEP257 have the leading triple quotes. Except that there is maybe an insinuation that the leading triple quotes is more like the "normal form" that docstring processing tools should create, not the way source code should be written. The normal-form-producing code is even provided verbatim in the text of PEP257.

Reply:
(I'm talking about the lack of newline after the first `"""`, not distance between a "one liner" and "definition".)

Reply:
I pattern it off the examples and my personal preference to maintain a close physical continuity between the brief description of the method with the method definition and its parameters/typestrings.

Reply:
Oh, I think I understand now. I thought by "close" you were referring to proximity between 
```
"""Load from a precomputed JSON artifact in the data directory.
```
and
```
Note: CountsProvider never loads centroids because it does not need them
"""
```
In both forms this proximity is 2 newlines:

```
"""Load from a precomputed JSON artifact in the data directory.

Note: CountsProvider never loads centroids because it does not need them
"""
```

```
"""
Load from a precomputed JSON artifact in the data directory.

Note: CountsProvider never loads centroids because it does not need them
"""
```
So by "one liner" you meant "the (first line of the) docstring" and by "definition" you meant "the call signature expression" (not the method definition).

This reasoning makes sense. I will use this style from now on.

Reply:
I pushed a commit implementing this style. I used a short python script to regex replace, it looked right in a sample of examples but I'm not 100% sure every docstring was treated corrrectly by my script.